src/sage/doctest/forker.py

```diff
@@ -830,7 +832,10 @@ def compiler(example):
         self.optionflags = original_optionflags
 
         # Record and return the number of failures and tries.
-        self._DocTestRunner__record_outcome(test, failures, tries)
+        try:
+            self._DocTestRunner__record_outcome(test, failures, tries, walltime_skips)
+        except TypeError:
+            self._DocTestRunner__record_outcome(test, failures, tries)
```

Comment:
This looks somewhat dangerous… because if a `TypeError` is raised within the function itself for some reason, the second is retried. What about using `inspect.signature` or `sys.version`?

(but then in that hypothetical case, the second would immediately error out anyway. Probably not a big deal.)

Reply:
In that case, the current behavior is unchanged, isn't it?

Reply:
Using `sys.version` (here and also above) would make it clear that the code can be cleaned up when python 3.12 support is deprecated.